packages/adblocker-extended-selectors/src/eval.ts

```diff
@@ -71,104 +71,238 @@ export function matches(element: Element, selector: AST): boolean {
       }
 
       return text.length >= minLength;
+    } else if (selector.name === 'matches-path') {
+      const { argument } = selector;
+      if (argument === undefined) {
+        return false;
+      }
+
+      const path = globalThis.window.location.pathname;
+
+      let pattern = argument;
+      if (pattern.startsWith('/') && pattern.endsWith('/')) {
+        pattern = pattern.slice(1, -1);
+      }
+
+      const regex = new RegExp(pattern);
+      return regex.test(path);
+    } else if (selector.name === 'matches-attr') {
+      const { argument } = selector;
+      if (argument === undefined) {
+        return false;
+      }
+
+      const indexOfEqual = argument.indexOf('=');
+      let namePattern, valuePattern;
+      if (indexOfEqual === -1) {
+        namePattern = argument;
+      } else {
+        namePattern = argument.slice(0, indexOfEqual);
+        valuePattern = argument.slice(indexOfEqual + 1);
+      }
+
+      let attrName = namePattern;
+      if (namePattern.startsWith('/') && namePattern.endsWith('/')) {
+        const regex = new RegExp(namePattern.slice(1, -1));
+        for (const attr of element.attributes) {
+          if (regex.test(attr.name)) {
+            attrName = attr.name;
+            break;
+          }
+        }
+      }
+
+      if (!valuePattern) {
+        return element.hasAttribute(attrName);
+      }
+
+      const value = element.getAttribute(attrName);
+
+      if (value === null) {
+        return false;
+      }
```

Comment:
Since the comment was lost, here again (more concrete with code, but not tested). It is an attempt to make the matching more explicit; not relying on the fallthrough when the regular expression does not match:

```suggestion
      let value;
      if (namePattern.startsWith('/') && namePattern.endsWith('/')) {
        const regex = new RegExp(namePattern.slice(1, -1));
        const elem = [...element.attributes].find(({ name }) => regex.test(name));
        if (elem === undefined) {
          return false;
        }
        value = elem.value;
      } else {
        value = element.getAttribute(valuePattern);
        if (value === null) {
          return false;
        }
      }

      if (!valuePattern) {
        return true;
      }
```